### PR TITLE
Test for multiple redirects

### LINKED ISSUE #18: Add tests for multiple redirects
ethers-ccip-read-provider should have tests checking that multiple ccip read redirects work, and that the correct error is returned if there are too many.

### PR DESCRIPTION
**Pull request by mdtanrikulu**
Resolves #18

### CODE REVIEW

File: packages/examples/trusted-gateway-token/contracts/contracts/TestMaxRetry.sol [hunk 1]
```diff
@@ -0,0 +1,47 @@
+//SPDX-License-Identifier: MIT
```
Comment (makoto) [inline packages/examples/trusted-gateway-token/contracts/contracts/TestMaxRetry.sol:1]: Is this file in use? Looks like duplicate

**Comment by makoto** [comment]
I don't see any code or config that enables "Too many redirect". Is this the default behavior of ethers.js?

File: packages/examples/trusted-gateway-token/contracts/contracts/TestMaxRetry.sol (hunk 1 above)
Comment (mdtanrikulu) [reply packages/examples/trusted-gateway-token/contracts/contracts/TestMaxRetry.sol:1]: the other contracts were in both places, for the consistency I placed this one too.

**Comment by mdtanrikulu** [comment]
> I don't see any code or config that enables "Too many redirect". Is this the default behavior of ethers.js?

it's a default behavior of both [ethers.js](https://github.com/ethers-io/ethers.js/blob/56fabe987bb8c1e4891fdf1e5d3fe8a4c0471751/packages/providers/src.ts/base-provider.ts#L1625) (10 max retry) and [ccip-read-provider](https://github.com/smartcontractkit/ccip-read/blob/9cc5816b19a645a26258ec5d759169eaa03368c3/packages/ethers-ccip-read-provider/src/index.ts#L96) (4 max retry)

**Review by makoto — APPROVED** [review]

**Pull request merged**